packages/block-library/src/post-tags/edit.js

```diff
@@ -34,9 +40,22 @@ function PostTagsDisplay() {
 	);
 }
 
-export default function PostTagsEdit() {
-	if ( ! useEntityId( 'postType', 'post' ) ) {
-		return __( 'Post Tags' );
+export default function PostTagsEdit( { context } ) {
+	if ( ! context.postType || ! context.postId ) {
+		return (
+			<Warning>
+				{ __( 'Post tags block: No post found for this block.' ) }
+			</Warning>
+		);
+	} else if ( context.postType !== 'post' ) {
+		return (
+			<Warning>
+				{ __(
+					'Post tags block: Tags are only available in posts. Please add this block to a post instead.'
```

Comment:
I like this second warning, I want to steal it for the Post Comments block as well.  Although Im not sure if it makes sense in either case given what @ockham was mentioning about plugins overriding the behavior of allowing tags (and possibly comments?) on non-post entities. 🤔

Reply:
We could maybe rephrase it a bit?

```suggestion
					'Post tags block: Tags are not available for this post type.'
```

Reply:
> We could maybe rephrase it a bit?

Post type makes sense as a developer. Are users generally familiar with the term `post type` as well? If they are, then I think the phrasing is great and I'll modify the text. 👍

If not, then maybe something more user friendly would be "Tags are not available for this view."

Reply:
Ah yeah, good point. Maybe not '... for this view.' since ideally the copy we'll choose will still ring familiar with people who know what a post type is. '... for this content' seems good. '... for this type of content.' maybe?

Reply:
Since plugins can allow tags for other CPTs maybe its best to just get rid of this second check?  After all, it should fall back to the 'No Tags.' copy in `PostTagsDisplay` when none are found? 🤔

Reply:
:man_shrugging: idk. For the uninitiated (that most likely don't have a plugin like that installed), it could be helpful to have messaging that makes it clear that they _can't_ have tags (so they don't go looking after just seeing 'No Tags.').

(But this discussion might be moot if we remove the block from the inserter anyway, as just [greenlit](https://github.com/WordPress/gutenberg/pull/24082#issuecomment-662657939) by @dubielzyk?)